### DISCUSSION TITLE
Load image on cognition.run

### DISCUSSION THREAD

**Discussion by ChongZhao-neuro** [jspsych/jsPsych#2290] [Q&A]
@javidalpe Thanks for creating cognition.run- I have created several experiments that involved the usage of "canvas-keyboard-response" and "image-keyboard-response" plus-ins. Prior to this weekend, our experiments seem to load pictures properly. However, I discovered today that all of my experiments involving these two plus-ins suddenly had issues loading the pictures.

1) The issue was there only when pictures were loaded, but not when we just have simple letters or background canvas using the plug-in.

2) I tried jspsych 6.1/6.2/6.3. Prior to this weekend everything seems to work properly on every of the three versions. However, now the pictures just won't load no matter which version we used.

My suspicion was that the upgrade from jspsych 6 to 7 last week may change some aspects of how the platform handles these plug-ins... I also tried to use preload to load images and then the page just stuck there forever... This never happened before either.

Thanks again for creating the platform and looking forward to your response!

**Comment by becky-gilbert** [comment]
Hi @ChongZhao-neuro, this is tricky to solve without looking at the code and seeing what's happening in the browser. Would you be willing to share a link to your experiment please? Or if you don't want to mess up your data collection, could you maybe clone the study or create a basic demo that reproduces the problem?

**Reply by ChongZhao-neuro** [reply]
Thanks Becky! I looked into the issue and finally found out that, under jspsych 6.3.0, if we use preload with the names of the pictures as input, the pictures would all be loaded on cognition.run. The autoload function just won't work on the code anymore, and if we changed it to anything other than 6.3.0 the preload will not work.

**Reply by becky-gilbert** [reply]
Hi @ChongZhao-neuro, it sounds like you might've figured out the problem? If so then that's great! But I'd still like to understand, just to clarify whether or not this a bug.

In jsPsych v6.3.0, we moved the preload settings from `jsPsych.init` into the new `preload` plugin. So if preloading is working for you in v6.3.0, then I guess you're doing something like this:
```js
var preload = {
  type: 'preload',
  images: ['img1.png','img2.png']
};
timeline.push(preload);
```
Is that right?

And when you say that the autoload function won't work on the code anymore, do you mean that the `preload` trial above doesn't work if you use the `auto_preload: true` parameter instead of specifying the images?

The `auto_preload: true` setting doesn't work in cases when the file isn't passed directly to an image/audio/video trial parameter - see [this section of the Media Preloading documentation](https://www.jspsych.org/7.0/overview/media-preloading/#manual-preloading) for an explanation and examples. Do you think that was the problem you're having? If not, it would be great if you could clarify so that we can understand the issue and make sure it's not a bug. Thanks!

**Comment by chiggei** [comment]
Hello, I'm experiencing a similar problem, that is, preloading the stimulus files takes very long and eventually fails, although there were no issues with the experiment up to a few days ago and the code has not been changed at any point. Also, in my case, using jspsych 6.3.0 instead of 6.3.1 (the version that I have been using) does not really change things. What does seem to change things is opening up the experiment on Firefox rather than Chrome, my default browser. There, the issue arises less often (but it still does arise sometimes). I didn't test other browsers. In general, though, Cognition seems to run slower these days. Any idea what may be happening?
By the way, I hate to show up only when I need to complain about Cognition, which has worked flawlessly these past few months for me and is such an amazing platform. I look forward to resuming collecting data with it!

**Reply by becky-gilbert** [reply]
Hi @chiggei, a few questions:
- are you seeing any errors in the browser's JavaScript console when the preloading fails?
- have you tried increasing the `preload` plugin's `max_load_time` parameter (see docs: https://www.jspsych.org/7.0/plugins/preload/#parameters)

Feel free to share a link to your experiment (or a copy of it, if you don't want to mess up counterbalancing etc.), and we can have a look.

**Reply by chiggei** [reply]
Hi Becky, I did see errors in the console when the issue occurred. If memory serves well, there were a bunch of 502 errors, one for each file failing to load. Unfortunately, I didn't think of copying and pasting those errors somewhere and now I cannot replicate the problem anymore, meaning that the problem is, apparently, gone (even on Chrome, which did not work a single time yesterday).
Also, I have not messed with the max_load_time parameter because I would be happy with there being infinite time for the files to load (the default option; I do have a bunch of files that need loading and my participants may not have the best internet connections). If there's reason to think that a deadline would help I would be happy to include it though (although I would still prefer a rather generous one).
Anyway, here's a link to a copy of the experiment: [https://jyd5woywzn.cognition.run](https://jyd5woywzn.cognition.run)
The issue used to show up right after the initial loading screen. The language of the task is Italian, but, if it allows you to proceed after the loading screen, it means that preloading went well. Thanks!
